src/bullsAndCows.js

```diff
@@ -25,8 +25,43 @@
  *
  * @return {object} - like {bulls: 0, cows: 0}
  */
+
 function bullsAndCows(generatedNumber, enteredNumber) {
-  // write code here
+  const BullsAndCows = {
+    'bulls': 0,
+    'cows': 0,
+  };
+
+  if (enteredNumber === undefined
+    || isRepeatSymbol(enteredNumber)
+    || enteredNumber.length !== generatedNumber.length
+    || isNaN(Number(enteredNumber))) {
+    return undefined;
+  }
+
+  for (let i = 0; i < generatedNumber.length; i++) {
+    if (generatedNumber[i] === enteredNumber[i]) {
```

Comment:
```
if (generatedNumber[i] === enteredNumber[i]) {
  BullsAndCows = {
    ...BullsAndCows,
    bulls: BullsAndCows.bulls + 1,
  }
} else if (generatedNumber.includes(enteredNumber[i])) {
  BullsAndCows = {
    ...BullsAndCows,
    cows: BullsAndCows.cows + 1,
  }
}
```
this is better so that you don't mutate the BullsAndCows object